app/controllers/admin/time_ranges_controller.rb

```diff
@@ -34,13 +35,17 @@ def update
   end
 
   def destroy
-    @time_range = TimeRange.find(params[:id])
     @time_range.destroy
     redirect_to admin_time_ranges_path, notice: notice('successfully.destroyed')
   end
 
   private
 
+  def find_time_range
+    @time_range = TimeRange.find(params[:id])
+    authorize @time_range
```

Comment:
if authorization is inside function find_time_range function name should verbosely say it or better put in in some after callback

Reply:
Could have used the before_action callback, but will always be doing both finding and authorizing at the same time for the same actions, so went with renaming the methods from find_resource to find_and_authorize_resource.